client/ayon_blender/api/ops.py

```diff
@@ -188,7 +189,22 @@ def _process_app_events() -> Optional[float]:
                 msg)
             dialog.setMinimumWidth(500)
             dialog.setDetailedText(detail)
-            dialog.exec_()
+            dialog.setAttribute(QtCore.Qt.WA_DeleteOnClose)
+            dialog.setStyleSheet(load_stylesheet())
+            # Ensure the dialog stays on top and is properly focused
+            dialog.setWindowFlags(
+                dialog.windowFlags() |
+                QtCore.Qt.WindowStaysOnTopHint |
+                QtCore.Qt.Dialog
+            )
+
+            try:
+                dialog.show()
+                dialog.raise_()
+                dialog.activateWindow()
+                dialog.exec_()
```

Comment:
@iLLiCiTiT this might be a Qt question - but in this case we want to show the modal dialog to the user - but don't need to respond on their action. It's just an "error occurred" message.

According to Qt docs, `exec()` is to be avoided [like stated here](https://doc.qt.io/qt-6/qdialog.html#exec) and instead we should use `dialog.open()`. Is this a case where `.open()` does make more sense then? (I've never used it myself.)

Reply:
Yep, `exec_` technically blocks the process, `open` just shows the message.